### PR TITLE
[TASK] Add support TYPO3 10.4 LTS

### LINKED ISSUE #11: Add support for TYPO3 10 X marks the spot

### PR DESCRIPTION
**Pull request by josefglatz**
Since the extension is actually feature-complete we decided to remove
support for TYPO3 < 10.

- delete removed ext_emconf.php settings and raise TYPO3 core
  version constraint
- remove TYPO3_MODE check in ext_localconf.php
- composer.json: use new spelling for used license
- composer.json: raise TYPO3 core version constraint
- composer.json: remove superfluous replace.gravatar option
- composer.json: add extra.typo3/cms.extension-key option
- use new ExtensionConfiguration class to retrieve extConf
- use PHP return type declarations
- ProxyController: use Request/Response
- ProxyController: remove bundled guzzle client logic
- remove bundled guzzle since it is shipped with TYPO3 core
- GravatarProvider: add missing default setting for useProxy
- README: add info about versions
- a11y: remove title attribute in extension icon

Resolves: #11

### CODE REVIEW

File: ext_emconf.php [hunk 1]
```diff
@@ -4,16 +4,13 @@
...
     'clearCacheOnLoad' => 1,
     'author' => 'Frans Saris',
     'author_email' => 'franssaris@gmail.com',
-    'author_company' => '',
-    'version' => '1.0.2',
+    'version' => '2.0.0',
     'constraints' => [
         'depends' => [
-            'typo3' => '7.5.0-9.5.99',
+            'typo3' => '10.4.0-10.4.99',
```
Comment (georgringer) [inline ext_emconf.php:13]:
```suggestion
            'typo3' => '10.4.0-10.5.99',
```

Comment (georgringer) [reply ext_emconf.php:13]: sorry BS :(

Comment (josefglatz) [reply ext_emconf.php:13]: April 1st is already over ;P

**Comment by fsaris** [comment]
tnx for the PR :tada:

**Pull request merged**